flow/controllers/car_following_models.py

```diff
@@ -381,11 +381,22 @@ def get_accel(self, env):
         v = env.vehicles.get_speed(self.veh_id)
         lead_id = env.vehicles.get_leader(self.veh_id)
         h = env.vehicles.get_headway(self.veh_id)
-
-        # negative headways may be registered by sumo at intersections/
-        # junctions. Setting them to 0 causes vehicles to not move; therefore,
-        # we maintain these negative headways to let sumo control the dynamics
-        # as it sees fit at these points.
+        edge = env.vehicles.get_edge(self.veh_id)
+        lead_edge = env.vehicles.get_edge(lead_id)
+
+        # This covers situations where both vehicles are in a junction from
+        # either direction (e.g. in the figure eight). The vehicle that is
+        # deeper into the junction has its accelerations defined by sumo. This
+        # needed is due to modifications in sumo from 1d4338ab80 to b6f37dd
```

Comment:
commit number in this line does not seem correct? should be 016c09d306

Reply:
Also, why would you want only the lead vehicle to have its acceleration defined by sumo? Why not both?

Reply:
As in have the acceleration of vehicles in a junction be defined by sumo? I can try that

Reply:
I mean, you're already doing it for the lead vehicle?

Reply:
I actually like your idea better, the solution is more cleaner and more elegant, and the behavior is the same. Will push changes now